service/service_test.go

```diff
@@ -0,0 +1,174 @@
+package service
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"testing"
+
+	"github.com/Diegoplas/go-bootcamp-deliverable/model"
+)
+
+type mockGetter struct {
+	listPokemonResponse []model.PokemonData
+	wantErr             bool
+}
+
+func (mg mockGetter) ListPokemons() ([]model.PokemonData, error) {
+	if mg.wantErr {
+		return nil, fmt.Errorf("list pokemons error")
+	}
+	return mg.listPokemonResponse, nil
+}
+
+func TestGetPokemonService_GetPokemonFromCSV(t *testing.T) {
+
+	tests := []struct {
+		name         string
+		mockedGetter mockGetter
+		requestedID  string
+	}{
+		{
+			name:         "Valid test",
+			mockedGetter: mockGetter{listPokemonResponse: MockedPokemonResponse},
+			requestedID:  "95",
+		},
+		{
+			name:         "Invalid - empty param",
+			mockedGetter: mockGetter{wantErr: true},
+			requestedID:  "",
+		},
+		{
+			name:         "Invalid - error on CSVdata functions",
+			mockedGetter: mockGetter{wantErr: true},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testRepo := NewRepositoryService(tt.mockedGetter)
+			gotPokemon, err := testRepo.GetPokemonFromCSV(tt.requestedID)
+			if (err != nil) != tt.mockedGetter.wantErr {
+				t.Errorf("GetPokemonFromCSV() error = %v, wantErr %v", err, tt.mockedGetter.wantErr)
+				return
+			}
+			gotID := ""
+			if gotPokemon.ID != 0 {
+				gotID = strconv.Itoa(gotPokemon.ID)
+			}
+			if gotID != tt.requestedID {
+				t.Errorf("GetPokemonFromCSV() Got ID = %v, wanted ID %v", gotID, tt.requestedID)
+			}
+		})
+	}
+}
+
+type mockRequestSender struct {
+	method   string
+	url      string
+	response *http.Response
+	wantErr  bool
+}
+
+type mockRequestParser struct {
```

Comment:
Is this mock actually being used? I don't see you using this anywhere

Reply:
I'm guessing you planned on using a requestSender and parser as a dependency right? I mean, this is the only place where I see this being declared but not used